api/v1alpha1/backend_types.go

```diff
@@ -135,6 +143,22 @@ type BackendSpec struct {
 	Fallback *bool `json:"fallback,omitempty"`
 }
 
+// BackendType defines the type of the Backend.
+type BackendType string
+
+const (
+	// BackendTypeEndpoints defines the type of the backend as Endpoints.
+	BackendTypeEndpoints BackendType = "Endpoints"
+	// BackendTypeDynamicResolver defines the type of the backend as DynamicResolver.
+	//
+	// When a backend is of type DynamicResolver, the Envoy will resolve the upstream
+	// ip address and port from the host header of the incoming request. If the ip address
+	// is directly set in the host header, the Envoy will use the ip address and port as the
+	// upstream address. If the hostname is set in the host header, the Envoy will resolve the
+	// ip address and port from the hostname using the DNS resolver.
+	BackendTypeDynamicResolver BackendType = "DynamicResolver"
```

Comment:
throwing some more options -
1)  "DynamicResolver"
2) "DynamicForwardProxy"
3) "HostResolver"
4) "ForwardProxy"

ptal @envoyproxy/gateway-maintainers @envoyproxy/gateway-reviewers

Reply:
+1 DynamicResolver or HostResolver

Reply:
+1 for DynamicResolver

Reply:
I'm fine with DynamicResolver. I don't think that any of the options expresses well what this does, which is to use an endpoint from the incoming request. I don't have a good idea for a name that expresses this.

Reply:
yah same, lets pick DynamicResolver for now then